pkg/operator/certrotation/client_cert_rotation_controller.go

```diff
@@ -44,23 +70,21 @@ type CertRotationController struct {
 	RotatedSelfSignedCertKeySecret RotatedSelfSignedCertKeySecret
 
 	// Plumbing:
-	OperatorClient v1helpers.StaticPodOperatorClient
+	StatusReporter StatusReporter
 }
 
 func NewCertRotationController(
-	name string,
```

Comment:
Suggest keeping name for event reporting (probably should have been there.

Reply:
The event recorder currently has `recorder.WithComponentSuffix("cert-rotation-controller")` - would you like to see the name as a further suffix on top?

Reply:
> The event recorder currently has `recorder.WithComponentSuffix("cert-rotation-controller")` - would you like to see the name as a further suffix on top? Or replace the hard-coded suffix entirely?

Your call, but I think having a distinction there matters.